### PR TITLE
create an endpoint for the hello function and a health check endpoint tested the end points with postman

### PR DESCRIPTION
**Pull request by MujtabaNasir**
create an endpoint for the hello function and a health check endpoint
tested the end points with postman

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,48 @@
...
+```
+
+3. To activate virtual environment:
+```bash
+. fastapi_env/Scripts/activate
+```
+
+4. To deactivate go into the folder where .bat files exists:
+```bash
+cd .\fastapi_env\Scripts\
```
Comment (mdsrqbl) [inline README.md]: use forward slashes

File: README.md [hunk 2]
```diff
@@ -1 +1,48 @@
...
+run command: deactivate
```
Comment (mdsrqbl) [inline README.md]: avoid writing instructions inside code blocks

File: README.md [hunk 3]
```diff
@@ -1 +1,48 @@
...
+```
+
+5. Install necessary dependancies
+```bash
+pip install fastapi uvicorn
```
Comment (mdsrqbl) [inline README.md]: use `-r requirements.txt` istead

File: README.md [hunk 4]
```diff
@@ -1 +1,48 @@
...
+```
+
+6. Install mujtaba-charm package:
+```bash
+pip install git+https://github.com/MujtabaNasir/mujtaba-charm.git
```
Comment (mdsrqbl) [inline README.md:41]: add to your requirements file

**Review by mdsrqbl — CHANGES_REQUESTED** [review]

**Comment by mdsrqbl** [comment]
integrate python logging into your service

File: README.md (hunk 1 above)
Comment (MujtabaNasir) [reply README.md]: Done

File: README.md (hunk 2 above)
Comment (MujtabaNasir) [reply README.md]: Done

File: README.md (hunk 3 above)
Comment (MujtabaNasir) [reply README.md]: Done

File: README.md (hunk 4 above)
Comment (MujtabaNasir) [reply README.md:41]: Done

File: main.py [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    """
+    Prints a greeting message.
+
+    Returns:
+        dict: A dictionary containing a greeting message.
+    """
+    logger.info("Handling /hello request")
+    message = hello()
+    logger.info(f"Returning message: {message}")
+    return {"message": message}
```
Comment (mdsrqbl) [inline main.py]: this is code duplication

File: main.py [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+@app.get("/hello/{name}")
+async def get_hello(name):
```
Comment (mdsrqbl) [inline main.py]: make the parameter optional to handle thee senario where parameter isn't passed

File: main.py [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+@app.get("/health")
+async def health_check():
+    """
+    Returns the health check status of the API.
+    """
+    logger.info("Handling /health request")
+    return {"status": "ok"}
+
+if __name__=="__main__":
+    uvicorn.run(app,host="0.0.0.0",port=8000)
```
Comment (mdsrqbl) [inline main.py]: format the code properly
(use black)

File: main.py [hunk 8]
```diff
@@ -0,0 +1,49 @@
+from fastapi import FastAPI
+import uvicorn
+from mujtaba_charm.utils.sample import hello
+import logging
```
Comment (mdsrqbl) [inline main.py]: sort your imports properly

**Review by mdsrqbl — CHANGES_REQUESTED** [review]

File: main.py [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+
+app = FastAPI()
+
+logging.basicConfig(
+    level=logging.INFO, format="[%(asctime)s] %(levelname)s in %(module)s: %(message)s"
+)
+logger = logging.getLogger(__name__)
+
+
+@app.get("/hello")
```
Comment (mdsrqbl) [inline main.py:15]: do you really need to write this endpoint twice?

File: main.py [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+
+    Returns:
+        dict: A dictionary containing a greeting message.
+    """
+    if name:
+        logger.info(f"Handling /hello/{name} request")
+        message = hello(name)
+    else:
+        logger.info("Handling /hello request")
+        message = hello()
```
Comment (mdsrqbl) [inline main.py]: remove code duplication

explore python's `or` operator to avoid the if/else

```python
hello(name or "my-dear-user")
```

**Review by mdsrqbl — CHANGES_REQUESTED** [review]

File: main.py (hunk 10 above)
Comment (MujtabaNasir) [reply main.py]: Done

File: main.py (hunk 9 above)
Comment (MujtabaNasir) [reply main.py:15]: @app.get("/hello") outputs "Hello World!"
with the same endpoint it outputs "Hello {name}!" with the browser url of http://127.0.0.1:8000/hello?name=Jack

**Review by mdsrqbl — APPROVED** [review]

**Pull request merged**